### PR TITLE
Universal Perturbations

### PR DESCRIPTION
**Pull request by dxoigmn**
# What does this PR do?

A big missing feature of MART is the ability to generate universal perturbations. This PR tries to change that.

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

## Before submitting

- [ ] The title is **self-explanatory** and the description **concisely** explains the PR
- [ ] My **PR does only one thing**, instead of bundling different changes together
- [ ] I list all the **breaking changes** introduced by this pull request
- [ ] I have commented my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run pre-commit hooks with `pre-commit run -a` command without errors

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: mart/attack/perturber.py [hunk 1]
```diff
@@ -103,4 +108,12 @@ def forward(self, **batch):
...
         input_adv = self.composer(self.perturbation, **batch)
 
-        return input_adv
+        # FIXME: This is a hack
+        total_variation = torch.mean(
+            torch.sum(torch.square(self.perturbation[:, 1:, :] - self.perturbation[:, :-1, :]))
+            + torch.sum(  # noqa: W503
+                torch.square(self.perturbation[:, :, 1:] - self.perturbation[:, :, :-1])
+            )
+        )
```
Comment (dxoigmn) [inline mart/attack/perturber.py:117]: What if I want to do an l2 loss instead? How do we make this more generic?

File: mart/nn/nn.py [hunk 2]
```diff
@@ -237,8 +237,16 @@ def __init__(self, *args, **kwargs):
 
 # FIXME: This must exist already?!
 class Sum(torch.nn.Module):
-    def __init__(self):
+    def __init__(self, weights=None):
```
Comment (dxoigmn) [inline mart/nn/nn.py:240]: Make this a parameter of forward.
